src/core/ext/client_channel/parse_address.c

```diff
@@ -49,12 +49,12 @@
 
 int parse_unix(grpc_uri *uri, grpc_resolved_address *resolved_addr) {
   struct sockaddr_un *un = (struct sockaddr_un *)resolved_addr->addr;
-
+  const size_t maxlen = sizeof(un->sun_path);
+  const size_t path_len = strnlen(uri->path, maxlen);
```

Comment:
Doesn't look like the logic actually changed here.  But upon further review, I think this is correct: if the NUL byte is at maxlen, strnlen() will return maxlen - 1, which is fine, and if the NUL byte is at maxlen + 1, we don't have enough room in the buffer, so we should fail.